src/deepsparse/server/main.py

```diff
@@ -0,0 +1,283 @@
+        results = await execute_async(
+            pipeline_def.pipeline, **vars(request), **pipeline_def.kwargs
+        )
+        return serializable_response(results)
+
+    if pipeline_def.config.alias:
+        # add the prediction path under the given alias
+        app.post(
+            f"/predict/{pipeline_def.config.alias}",
+            response_model=pipeline_def.response_model,
+            tags=["prediction"],
+        )(_predict_func)
+    elif num_models == 1:
+        # add only a single predict route since this is the only model we're serving
+        app.post(
+            "/predict", response_model=pipeline_def.response_model, tags=["prediction"]
+        )(_predict_func)
+    elif pipeline_def.config.task not in defined_tasks:
+        # fall back on adding the model to the task provided nothing is already assigned
+        app.post(
+            f"/predict/{pipeline_def.config.task}",
+            response_model=pipeline_def.response_model,
+            tags=["prediction"],
+        )(_predict_func)
+        defined_tasks.add(pipeline_def.config.task)
+    else:
+        raise ValueError(
+            f"Multiple tasks defined for {pipeline_def.config.task} and no alias "
+            f"given for {pipeline_def.config}. "
+            "Either define an alias or supply a single model for the task"
+        )
+
+
+def server_app_factory():
+    """
+    :return: a FastAPI app initialized with defined routes and ready for inference.
+        Use with a wsgi server such as uvicorn.
+    """
+    app = FastAPI(
+        title="deepsparse.server",
+        version=version,
+        description="DeepSparse Inference Server",
+    )
+    _LOGGER.info("created FastAPI app for inference serving")
+
+    config = server_config_from_env()
+    initialize_aysnc(config.workers)
+    _LOGGER.debug("loaded server config %s", config)
+    _add_general_routes(app, config)
+
+    pipeline_defs = load_pipelines_definitions(config)
+    _LOGGER.debug("loaded pipeline definitions from config %s", pipeline_defs)
+    num_tasks = len(config.models)
+    defined_tasks = set()
+    for pipeline_def in pipeline_defs:
+        _add_pipeline_route(app, pipeline_def, num_tasks, defined_tasks)
+
+    return app
+
+
+@click.command()
+@click.option(
+    "--host",
+    type=str,
+    default="0.0.0.0",
+    help="Bind socket to this host. Use --host 0.0.0.0 to make the application "
+    "available on your local network. "
+    "IPv6 addresses are supported, for example: --host '::'. Defaults to 0.0.0.0",
+)
+@click.option(
+    "--port",
+    type=int,
+    default=5543,
+    help="Bind to a socket with this port. Defaults to 5543.",
+)
+@click.option(
+    "--workers",
+    type=int,
+    default=1,
+    help="Use multiple worker processes. Defaults to 1.",
+)
+@click.option(
+    "--log_level",
+    type=str,
+    default="info",
+    help="Bind to a socket with this port. Defaults to info.",
```

Comment:
do we want to add choices here, or just let it fail?

Reply:
yep, added